test/test_support.ts

```diff
@@ -143,3 +147,178 @@ export function goldenTests(): GoldenFileTest[] {
+  const lines = source.split('\n');
+  const line = lines.findIndex(l => l.indexOf(token) !== -1) + 1;
+  if (line === 0) {
+    throw new Error(`Couldn't find token '${token}' in source`);
+  }
+  const column = lines[line - 1].indexOf(token);
+  return {line, column};
+}
+
+export interface CompilerOptions {
+  useTransformer: boolean;
+  outFile?: string;
+  filesNotToProcess: Set<string>;
+  inlineSourceMap: boolean;
+  generateDTS: boolean;
+}
+
+const DEFAULT_COMPILER_OPTIONS = {
+  filesNotToProcess: new Set<string>(),
+  inlineSourceMap: false,
+  generateDTS: false,
+};
+
+export function compile(
+    sources: Map<string, string>,
+    partialOptions: Partial<CompilerOptions>&{useTransformer: boolean}): {
+  compiledJS: string,
+  dts: string | undefined,
+  sourceMap: SourceMapConsumer,
+  sourceMapText: string
+} {
+  const options: CompilerOptions = {...DEFAULT_COMPILER_OPTIONS, ...partialOptions};
+  const resolvedSources = new Map<string, string>();
+  for (const fileName of toArray(sources.keys())) {
+    resolvedSources.set(ts.sys.resolvePath(fileName), sources.get(fileName)!);
+  }
+
+  let compilerOptions: ts.CompilerOptions;
+  if (options.inlineSourceMap) {
+    compilerOptions = {
+      inlineSourceMap: options.inlineSourceMap,
+      inlineSources: true,
+      outFile: options.outFile,
+      experimentalDecorators: true,
+      declaration: options.generateDTS,
+    };
+  } else {
+    compilerOptions = {
+      sourceMap: true,
+      inlineSources: true,
+      outFile: options.outFile,
+      experimentalDecorators: true,
+      declaration: options.generateDTS,
+    };
+  }
+
+  const fileNames = toArray(sources.keys());
+
+  const tsickleHost: tsickle.TsickleHost&tsickle.TransformerHost = {
+    shouldSkipTsickleProcessing: (fileName) =>
+        fileNames.indexOf(fileName) === -1 || options.filesNotToProcess.has(fileName),
+    pathToModuleName: cliSupport.pathToModuleName,
+    shouldIgnoreWarningsForPath: (filePath) => false,
+    fileNameToModuleId: (fileName) => fileName,
+  };
+
+  const {jsFiles, diagnostics} = options.useTransformer ?
+      compileWithTransfromer(resolvedSources, fileNames, tsickleHost, compilerOptions) :
+      compileWithTsickleCompilerHost(resolvedSources, fileNames, tsickleHost, compilerOptions);
+
+  if (diagnostics.length) {
+    console.error(tsickle.formatDiagnostics(diagnostics));
+    assert.fail();
+    return {compiledJS: '', dts: '', sourceMap: sourceMapTextToConsumer(''), sourceMapText: ''};
+  }
+
+  let outFile = options.outFile;
+  if (!outFile) {
+    if (sources.size === 1) {
+      const inputFileName = toArray(sources.keys())[0];
+      outFile = inputFileName.substring(0, inputFileName.length - 3) + '.js';
+    } else {
+      outFile = 'input.js';
+    }
+  }
+  const compiledJS = getFileWithName(outFile, jsFiles);
+
+  if (!compiledJS) {
+    assert.fail();
+    return {compiledJS: '', dts: '', sourceMap: sourceMapTextToConsumer(''), sourceMapText: ''};
```

Comment:
`assert.fail()` doesn't return, so this line is misleading. If the type system requires you to prove compiledJS is non-null, then `throw new Error('unreachable')`, otherwise just remove the line.